File: CmsAdmin/user_content/app/dtos/account_info_dto.py
from datetime import date
from dataclasses import dataclass
from typing import List


@dataclass
class MediaDto:
    url: str
    type: int


@dataclass
class AccountInfoDto:
    id: str
    full_name: str
    avatar: str
    birth_date: date
    gender: int
    medias: List[MediaDto]
    bio: str = ""
    address: str = ""
    job: str = ""
    reason: str = ""

File: interactive_main_service/src/auth_management/app/gateways/dtos/accountInfo.dto.ts
export class AccountInfoDto {
  id: string;
  fullName: string;
  avatar: string;
  birthDate: Date;
  gender: number;
  bio: string;
  address: string;
  job: string;
  reason: string;

  constructor(id: string, fullName: string, avatar: string, birthDate: Date, gender: number, bio = '', address = '', job = '', reason = '') {
    this.id = id;
    this.fullName = fullName;
    this.avatar = avatar;
    this.birthDate = birthDate;
    this.gender = gender;
    this.bio = bio;
    this.address = address;
    this.job = job;
    this.reason = reason;
  }
}
